widget/entry-with-completion.go

```diff
@@ -0,0 +1,140 @@
+func NewCompletionEntry(options []string) *CompletionEntry {
+	c := &CompletionEntry{items: options}
+	c.ExtendBaseWidget(c)
+	return c
+}
+
+// ShowCompletion displays the completion menu
+func (c *CompletionEntry) ShowCompletion() {
+	if c.pause {
+		return
+	}
+
+	if c.navigableList == nil {
+		c.navigableList = newNavigableList(c.items, &c.Entry, c.setTextFromMenu, c.HideCompletion)
+	}
+	holder := fyne.CurrentApp().Driver().CanvasForObject(c)
+	pos := fyne.CurrentApp().Driver().AbsolutePositionForObject(c)
+
+	if c.popupMenu == nil {
+		c.popupMenu = widget.NewPopUp(c.navigableList, holder)
+	}
+	max := fyne.Min(10, float32(len(c.items)))
+	c.popupMenu.Resize(fyne.NewSize(c.Entry.Size().Width, c.popupMenu.MinSize().Height*max-3*theme.Padding()))
+	c.popupMenu.ShowAtPosition(fyne.Position{X: pos.X, Y: pos.Y + c.Size().Height})
+	holder.Focus(c.navigableList)
+}
+
+// HideCompletion hides the completion menu.
+func (c *CompletionEntry) HideCompletion() {
+	if c.popupMenu != nil {
+		c.popupMenu.Hide()
+	}
+}
+
+// SetOptions set the completion list with itemList.
+func (c *CompletionEntry) SetOptions(itemList []string) {
+	c.items = itemList
+	if c.navigableList != nil {
+		c.navigableList.SetOptions(itemList)
+	}
+}
+
+// Prevent the menu to open when the user validate value from the menu.
+func (c *CompletionEntry) setTextFromMenu(s string) {
+	c.pause = true
+	c.Entry.SetText(s)
+	c.Entry.Refresh()
+	c.pause = false
+	c.popupMenu.Hide()
+}
+
+type navigableList struct {
+	*widget.List
+	data            binding.StringList
+	entry           *widget.Entry
+	selected        int
+	setTextFromMenu func(string)
+	hide            func()
+	navigating      bool
+}
+
+func newNavigableList(items []string, entry *widget.Entry, setTextFromMenu func(string), hide func()) *navigableList {
+	n := &navigableList{entry: entry, selected: -1, setTextFromMenu: setTextFromMenu, hide: hide}
+	n.data = binding.BindStringList(&items)
+
+	n.List = widget.NewListWithData(n.data, func() fyne.CanvasObject {
+		return widget.NewLabel("")
+	}, func(i binding.DataItem, o fyne.CanvasObject) {
+		o.(*widget.Label).Bind(i.(binding.String))
+	})
+
+	n.OnSelected = func(id int) {
+		if !n.navigating {
+			item, _ := n.data.GetItem(id)
+			val, _ := item.(binding.String).Get()
+			setTextFromMenu(val)
+		}
+		n.navigating = false
+	}
+
+	return n
+}
+
+func (n *navigableList) SetOptions(items []string) {
+	n.Unselect(n.selected)
+	n.data.Set(items)
+	n.selected = -1
+}
+
+func (n *navigableList) TypedKey(event *fyne.KeyEvent) {
+	n.entry.TypedKey(event)
+	switch event.Name {
+	case fyne.KeyDown:
+		if n.selected < n.data.Length()-1 {
+			n.selected++
+		}
+		n.navigating = true
+		n.List.Select(n.selected)
+
+	case fyne.KeyUp:
+		if n.selected > 0 {
+			n.selected--
+		} else {
+			n.Unselect(n.selected)
+			n.selected = -1
+		}
+		n.navigating = true
+		n.List.Select(n.selected)
```

Comment:
n.Select should be enough as you are embedding the list :)

Reply:
right, I'll do it in a few minutes